Mage.Sets/src/mage/cards/z/ZilorthaStrengthIncarnate.java

```diff
@@ -0,0 +1,82 @@
+package mage.cards.z;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.ContinuousEffect;
+import mage.abilities.effects.ContinuousEffectImpl;
+import mage.abilities.keyword.TrampleAbility;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.StaticFilters;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.permanent.ControllerIdPredicate;
+import mage.game.Game;
+
+import java.util.UUID;
+
+/**
+ * @author htrajan
+ */
+public final class ZilorthaStrengthIncarnate extends CardImpl {
+
+    public ZilorthaStrengthIncarnate(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{R}{G}");
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.DINOSAUR);
+        this.power = new MageInt(7);
+        this.toughness = new MageInt(3);
+
+        // Trample
+        this.addAbility(TrampleAbility.getInstance());
+
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new ZilorthaStrengthIncarnateEffect()));
+    }
+
+    private ZilorthaStrengthIncarnate(ZilorthaStrengthIncarnate card) {
+        super(card);
+    }
+
+    @Override
+    public Card copy() {
+        return new ZilorthaStrengthIncarnate(this);
+    }
+}
+
+class ZilorthaStrengthIncarnateEffect extends ContinuousEffectImpl {
+
+    ZilorthaStrengthIncarnateEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Benefit);
+        staticText = "Lethal damage dealt to creatures you control is determined by their power rather than their toughness";
+    }
+
+    private ZilorthaStrengthIncarnateEffect(ZilorthaStrengthIncarnateEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Layer layer, SubLayer sublayer, Ability source, Game game) {
+        // Change the rule
+        FilterCreaturePermanent filter = StaticFilters.FILTER_PERMANENT_CREATURE.copy();
+        filter.add(new ControllerIdPredicate(source.getControllerId()));
+        game.addPowerInsteadOfToughnessForDamageLethalityFilter(source.getSourceId(), filter);
```

Comment:
It adds only one time, but will be ignored on new filter for same source next time (see `putIfAbsent` usage in `addPowerInsteadOfToughnessForDamageLethalityFilter`). It can be potentially bugged and I recommends change it to simple `put`.

Reply:
yeah, fair enough, fixed